In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


In [2]:
df = pd.read_csv('../Datasets/Electric and Alternative Fuel Charging Stations.csv')

C:\Users\Usuario\AppData\Local\Temp\ipykernel_17636\3514369627.py:1: DtypeWarning: Columns (6,16,20,31,33,36,39,40,41,43,46,52,53,55,57,58,60,62) have mixed types. Specify dtype option on import or set low_memory=False.
  df = pd.read_csv('../Datasets/Electric and Alternative Fuel Charging Stations.csv')


In [3]:
df

Fuel Type Code                                  Station Name  \
0                CNG          Spire - Montgomery Operations Center   
1                CNG                           PS Energy - Atlanta   
2                CNG  Metropolitan Atlanta Rapid Transit Authority   
3                CNG                         United Parcel Service   
4                CNG                    Arkansas Oklahoma Gas Corp   
...              ...                                           ...   
70401           ELEC                     SCHLITZ PARK SCHLITZ PARK   
70402           ELEC                          NEXUS EAST STATION 2   
70403           ELEC                                       Raley's   
70404           ELEC                           13601 Glenoaks blvd   
70405           ELEC                            6723 Van Nuys blvd   

               Street Address  \
0            2951 Chestnut St   
1            340 Whitehall St   
2         2424 Piedmont Rd NE   
3        270 Marvin Miller Dr   
4           2100 S Waldron Rd   
...                       ...   
70401  1555 N River Center Dr   
70402        720 Airport Blvd   
70403   6119 Horseshoe Bar Rd   
70404     13601 Glenoaks blvd   
70405      6723 Van Nuys Blvd   

                                 Intersection Directions         City State  \
0                                                    NaN   Montgomery    AL   
1      From I-7585 N, exit 91 to Central Ave, left on...      Atlanta    GA   
2                                                    NaN      Atlanta    GA   
3                                                    NaN      Atlanta    GA   
4                                                    NaN   Fort Smith    AR   
...                                                  ...          ...   ...   
70401                                                NaN    Milwaukee    WI   
70402                                                NaN       Austin    TX   
70403                                                NaN       Loomis    CA   
70404                                                NaN  Los Angeles    CA   
70405                                                NaN  Los Angeles    CA   

         ZIP  Plus4 Station Phone Status Code  ... EV Pricing (French)  \
0      36107    NaN           NaN           E  ...                 NaN   
1      30303    NaN  770-350-3000           E  ...                 NaN   
2      30324    NaN           NaN           E  ...                 NaN   
3      30336    NaN           NaN           E  ...                 NaN   
4      72903    NaN  479-783-3188           E  ...                 NaN   
...      ...    ...           ...         ...  ...                 ...   
70401  53212    NaN  888-758-4389           E  ...                 NaN   
70402  78702    NaN  888-758-4389           E  ...                 NaN   
70403  95650    NaN  877-455-3833           E  ...                 NaN   
70404  91342    NaN  888-356-8911           E  ...                 NaN   
70405  91405    NaN  888-356-8911           E  ...                 NaN   

      LPG Nozzle Types Hydrogen Pressures Hydrogen Standards  \
0                  NaN                NaN                NaN   
1                  NaN                NaN                NaN   
2                  NaN                NaN                NaN   
3                  NaN                NaN                NaN   
4                  NaN                NaN                NaN   
...                ...                ...                ...   
70401              NaN                NaN                NaN   
70402              NaN                NaN                NaN   
70403              NaN                NaN                NaN   
70404              NaN                NaN                NaN   
70405              NaN                NaN                NaN   

      CNG Fill Type Code CNG PSI CNG Vehicle Class  LNG Vehicle Class  \
0                      B    3600                MD                NaN   
1              

In [4]:
# Obtener el conteo de nulos por columna de df
null_count = df.isnull().sum()

# Obtener el porcentaje de nulos por columna
null_percentage = (df.isnull().mean()) * 100

# Crear un nuevo DataFrame con la información de conteo y porcentaje
null_info = pd.DataFrame({
    'Nulos': null_count,
    'Porcentaje': null_percentage
})

# Imprimir el nuevo DataFrame
print(null_info.sort_values('Nulos', ascending=False))

                              Nulos  Porcentaje
Plus4                         70406  100.000000
BD Blends (French)            70404   99.997159
EV Other Info                 70357   99.930404
LNG On-Site Renewable Source  70344   99.911939
Hydrogen Status Link          70323   99.882112
...                             ...         ...
ID                                0    0.000000
Updated At                        0    0.000000
Station Name                      0    0.000000
Country                           0    0.000000
Fuel Type Code                    0    0.000000

[65 rows x 2 columns]


Filtrando por datos de vehiculos eléctricos

In [5]:
df = df[df['Fuel Type Code'] == 'ELEC']
df

Fuel Type Code                     Station Name          Street Address  \
156             ELEC         LADWP - Truesdale Center      11797 Truesdale St   
157             ELEC  LADWP - West LA District Office   1394 S Sepulveda Blvd   
158             ELEC    Los Angeles Convention Center      1201 S Figueroa St   
159             ELEC    LADWP - John Ferraro Building           111 N Hope St   
160             ELEC       LADWP - Haynes Power Plant           6801 E 2nd St   
...              ...                              ...                     ...   
70401           ELEC        SCHLITZ PARK SCHLITZ PARK  1555 N River Center Dr   
70402           ELEC             NEXUS EAST STATION 2        720 Airport Blvd   
70403           ELEC                          Raley's   6119 Horseshoe Bar Rd   
70404           ELEC              13601 Glenoaks blvd     13601 Glenoaks blvd   
70405           ELEC               6723 Van Nuys blvd      6723 Van Nuys Blvd   

        Intersection Directions         City State    ZIP  Plus4  \
156                         NaN   Sun Valley    CA  91352    NaN   
157                         NaN  Los Angeles    CA  90024    NaN   
158    West hall and South hall  Los Angeles    CA  90015    NaN   
159                 Across Hope  Los Angeles    CA  90012    NaN   
160                         NaN   Long Beach    CA  90803    NaN   
...                         ...          ...   ...    ...    ...   
70401                       NaN    Milwaukee    WI  53212    NaN   
70402                       NaN       Austin    TX  78702    NaN   
70403                       NaN       Loomis    CA  95650    NaN   
70404                       NaN  Los Angeles    CA  91342    NaN   
70405                       NaN  Los Angeles    CA  91405    NaN   

      Station Phone Status Code  ... EV Pricing (French) LPG Nozzle Types  \
156             NaN           E  ...                 NaN              NaN   
157             NaN           E  ...                 NaN              NaN   
158    213-741-1151           E  ...                 NaN              NaN   
159             NaN           E  ...                 NaN              NaN   
160             NaN           E  ...                 NaN              NaN   
...             ...         ...  ...                 ...              ...   
70401  888-758-4389           E  ...                 NaN              NaN   
70402  888-758-4389           E  ...                 NaN              NaN   
70403  877-455-3833           E  ...                 NaN              NaN   
70404  888-356-8911           E  ...                 NaN              NaN   
70405  888-356-8911           E  ...                 NaN              NaN   

      Hydrogen Pressures Hydrogen Standards CNG Fill Type Code CNG PSI  \
156                  NaN                NaN                NaN     NaN   
157                  NaN                NaN                NaN     NaN   
158                  NaN                NaN                NaN     NaN   
159                  NaN                NaN                NaN     NaN   
160                  NaN                NaN                NaN     NaN   
...                  ...                ...                ...     ...   
70401                NaN                NaN                NaN     NaN   
70402                NaN                NaN                NaN     NaN   
70403                NaN                NaN                NaN     NaN   
70404                NaN                NaN                NaN     NaN   
70405                NaN                NaN                NaN     NaN   

      CNG Vehicle Class  LNG Vehicle Class  EV On-Site Renewable Source  \
156                 NaN                NaN                          NaN   
157                 NaN                NaN                          NaN   
158                 NaN                NaN                          NaN   
159                 NaN                NaN                          NaN   
160                 NaN    

Filtrando por Estado de Nueva york

In [6]:
df[df['State'] == 'NY']

Fuel Type Code                                       Station Name  \
2087            ELEC  New York State Energy Research Development Aut...   
2977            ELEC                   James J Peters VA Medical Center   
3333            ELEC                                     Thruway Nissan   
3334            ELEC                                     Thruway Nissan   
3335            ELEC                               Country Club Imports   
...              ...                                                ...   
70266           ELEC                               GENESEO W PARK RIGHT   
70267           ELEC                              GENESEO W PARK CENTER   
70278           ELEC                           G & M REALTY 2 STATION 1   
70288           ELEC                             40 N 4th St Apartments   
70395           ELEC                                 EVolveNY Salamanca   

                     Street Address Intersection Directions       City State  \
2087                17 Columbia Cir                     NaN     Albany    NY   
2977           130 W Kingsbridge Rd        JJPVA Motor Pool      Bronx    NY   
3333                   79 Route 17K                     NaN   Newburgh    NY   
3334                   79 Route 17K                     NaN   Newburgh    NY   
3335                   55 Oneida St                     NaN    Oneonta    NY   
...                             ...                     ...        ...   ...   
70266  4271 Genesee Valley Plaza Rd                     NaN    Geneseo    NY   
70267  4271 Genesee Valley Plaza Rd                     NaN    Geneseo    NY   
70278             22-44 Jackson Ave                     NaN        LIC    NY   
70288                   40 N 4th ST                     NaN   Brooklyn    NY   
70395              768 Broad Street                     NaN  Salamanca    NY   

         ZIP  Plus4 Station Phone Status Code  ... EV Pricing (French)  \
2087   12203    NaN           NaN           E  ...                 NaN   
2977   10468    NaN           NaN           E  ...                 NaN   
3333   12550    NaN  845-562-1000           E  ...                 NaN   
3334   12550    NaN  845-562-1000           E  ...                 NaN   
3335   13820    NaN  607-432-2800           E  ...                 NaN   
...      ...    ...           ...         ...  ...                 ...   
70266  14454    NaN  888-758-4389           E  ...                 NaN   
70267  14454    NaN  888-758-4389           E  ...                 NaN   
70278  11101    NaN  888-758-4389           E  ...                 NaN   
70288  11249    NaN  888-998-2546           E  ...                 NaN   
70395  14779    NaN  855-900-7584           E  ...                 NaN   

      LPG Nozzle Types Hydrogen Pressures Hydrogen Standards  \
2087               NaN                NaN                NaN   
2977               NaN                NaN                NaN   
3333               NaN                NaN                NaN   
3334               NaN                NaN                NaN   
3335               NaN                NaN                NaN   
...                ...                ...                ...   
70266              NaN                NaN                NaN   
70267              NaN                NaN                NaN   
70278              NaN                NaN                NaN   
70288              NaN                NaN                NaN   
70395              NaN                NaN                NaN   

      CNG Fill Type Code CNG PSI CNG Vehicle Class  LNG Vehicle Class  \
2087                 NaN     NaN               NaN                NaN   
2977                 NaN     NaN               NaN                NaN   
3333                 NaN     NaN               NaN                NaN   
3334                 NaN     NaN               NaN                NaN   
3335                 NaN     NaN               NaN                NaN   
...                  ...     ...               .

In [7]:
def can_nulos():    
    # Obtener el conteo de nulos por columna de df
    null_count = df.isnull().sum()

    # Obtener el porcentaje de nulos por columna
    null_percentage = (df.isnull().mean()) * 100

    # Crear un nuevo DataFrame con la información de conteo y porcentaje
    null_info = pd.DataFrame({
        'Nulos': null_count,
        'Porcentaje': null_percentage
    })

    # Imprimir el nuevo DataFrame
    print(null_info.sort_values('Nulos', ascending=False))

In [8]:
print(can_nulos())

                              Nulos  Porcentaje
LPG Primary                   60907       100.0
E85 Other Ethanol Blends      60907       100.0
LNG On-Site Renewable Source  60907       100.0
Hydrogen Pressures            60907       100.0
CNG Storage Capacity          60907       100.0
...                             ...         ...
ID                                0         0.0
Updated At                        0         0.0
Station Name                      0         0.0
Country                           0         0.0
Fuel Type Code                    0         0.0

[65 rows x 2 columns]
None


Filtrando por Ciudad de Nueva York

In [9]:
df = df[df['City'] == 'New York']

In [10]:
df

Fuel Type Code                               Station Name  \
7777            ELEC                Port Authority Bus Terminal   
9645            ELEC                      United Parcel Service   
14384           ELEC                                   Spectrum   
14463           ELEC                 Central Parking - Tower 31   
14669           ELEC                               Tribeca Park   
...              ...                                        ...   
69770           ELEC                             DOCK STATION 9   
69771           ELEC                            DOCK STATION 10   
69823           ELEC                      Lincoln Square Garage   
69940           ELEC                              320 Mason Ave   
70215           ELEC  Delancey & Essex Municipal Parking Garage   

          Street Address                            Intersection Directions  \
7777         625 8th Ave                                                NaN   
9645       643 W 43rd St                                                NaN   
14384     401 W 219th St                                                NaN   
14463    9 W 31st Street                                                NaN   
14669    400 Chambers St  Residents & Visitors Only; See valet \nattenda...   
...                  ...                                                ...   
69770      1540 Broadway                                                NaN   
69771      1540 Broadway                                                NaN   
69823  178 W 69th Street  Entrance on 179 W 68th St or 184 W 69th \nSt b...   
69940      320 Mason Ave                                                NaN   
70215        107 Essx St                                                NaN   

           City State    ZIP  Plus4 Station Phone Status Code  ...  \
7777   New York    NY  10018    NaN  866-816-7584           E  ...   
9645   New York    NY  10036    NaN           NaN           E  ...   
14384  New York    NY  10034    NaN  866-816-7584           E  ...   
14463  New York    NY  10001    NaN  888-998-2546           E  ...   
14669  New York    NY  10282    NaN  888-998-2546           E  ...   
...         ...   ...    ...    ...           ...         ...  ...   
69770  New York    NY  10155    NaN  888-758-4389           E  ...   
69771  New York    NY  10001    NaN  888-758-4389           E  ...   
69823  New York    NY  10023    NaN  888-998-2546           E  ...   
69940  New York    NY  10305    NaN  888-356-8911           E  ...   
70215  New York    NY  10002    NaN  866-816-7584           E  ...   

      EV Pricing (French) LPG Nozzle Types Hydrogen Pressures  \
7777                  NaN              NaN                NaN   
9645                  NaN              NaN                NaN   
14384                 NaN              NaN                NaN   
14463                 NaN              NaN                NaN   
14669                 NaN              NaN                NaN   
...                   ...              ...                ...   
69770                 NaN              NaN                NaN   
69771                 NaN              NaN                NaN   
69823                 NaN              NaN                NaN   
69940                 NaN              NaN                NaN   
70215                 NaN              NaN                NaN   

      Hydrogen Standards CNG Fill Type Code CNG PSI CNG Vehicle Class  \
7777                 NaN                NaN     NaN               NaN   
9645                 NaN                NaN     NaN               NaN   
14384                NaN                NaN     NaN               NaN   
14463                NaN                NaN     NaN               NaN   
14669                NaN                NaN     NaN               NaN   
...                  ...                ...     ...               ...   
69770                NaN                NaN     NaN               NaN   
69771                NaN                NaN   

In [11]:
can_nulos()

                              Nulos  Porcentaje
LPG Nozzle Types                379       100.0
NG Fill Type Code               379       100.0
CNG Storage Capacity            379       100.0
LPG Primary                     379       100.0
LNG On-Site Renewable Source    379       100.0
...                             ...         ...
Updated At                        0         0.0
Station Name                      0         0.0
EV Connector Types                0         0.0
Country                           0         0.0
Open Date                         0         0.0

[65 rows x 2 columns]


Eliminando las columnas que tengan el 100 de los datos nulos

In [12]:
df = df.dropna(axis=1, how='all')
df

Fuel Type Code                               Station Name  \
7777            ELEC                Port Authority Bus Terminal   
9645            ELEC                      United Parcel Service   
14384           ELEC                                   Spectrum   
14463           ELEC                 Central Parking - Tower 31   
14669           ELEC                               Tribeca Park   
...              ...                                        ...   
69770           ELEC                             DOCK STATION 9   
69771           ELEC                            DOCK STATION 10   
69823           ELEC                      Lincoln Square Garage   
69940           ELEC                              320 Mason Ave   
70215           ELEC  Delancey & Essex Municipal Parking Garage   

          Street Address                            Intersection Directions  \
7777         625 8th Ave                                                NaN   
9645       643 W 43rd St                                                NaN   
14384     401 W 219th St                                                NaN   
14463    9 W 31st Street                                                NaN   
14669    400 Chambers St  Residents & Visitors Only; See valet \nattenda...   
...                  ...                                                ...   
69770      1540 Broadway                                                NaN   
69771      1540 Broadway                                                NaN   
69823  178 W 69th Street  Entrance on 179 W 68th St or 184 W 69th \nSt b...   
69940      320 Mason Ave                                                NaN   
70215        107 Essx St                                                NaN   

           City State    ZIP Station Phone Status Code  \
7777   New York    NY  10018  866-816-7584           E   
9645   New York    NY  10036           NaN           E   
14384  New York    NY  10034  866-816-7584           E   
14463  New York    NY  10001  888-998-2546           E   
14669  New York    NY  10282  888-998-2546           E   
...         ...   ...    ...           ...         ...   
69770  New York    NY  10155  888-758-4389           E   
69771  New York    NY  10001  888-758-4389           E   
69823  New York    NY  10023  888-998-2546           E   
69940  New York    NY  10305  888-356-8911           E   
70215  New York    NY  10002  866-816-7584           E   

                 Groups With Access Code  ...   Open Date  EV Connector Types  \
7777                              Public  ...  2018-04-01               J1772   
9645                             Private  ...  2016-03-01               J1772   
14384                             Public  ...  2016-06-27               J1772   
14463                             Public  ...  2014-08-20               J1772   
14669                            Private  ...  2014-11-12               J1772   
...                                  ...  ...         ...                 ...   
69770                             Public  ...  2022-07-12               J1772   
69771                             Public  ...  2022-07-12               J1772   
69823                             Public  ...  2022-07-12               J1772   
69940                             Public  ...  2022-07-15               J1772   
70215  Public - Credit card at all times  ...  2022-07-16  CHADEMO J1772COMBO   

       Country        Groups With Access Code (French) Access Code  \
7777        US                                  Public      public   
9645        US                                   Privé     private   
14384       US                                  Public      public   
14463       US                                  Public      public   
14669       US                                   Privé     private   
...        ...                                     ...         ...   
69770       US                                  Public      public   
69771       US        

In [13]:
can_nulos()

                                  Nulos  Porcentaje
Federal Agency ID                   375   98.944591
Federal Agency Code                 375   98.944591
Federal Agency Name                 375   98.944591
EV DC Fast Count                    370   97.625330
Access Detail Code                  364   96.042216
Intersection Directions             347   91.556728
Cards Accepted                      347   91.556728
Facility Type                       119   31.398417
Owner Type Code                     100   26.385224
EV Pricing                           88   23.218997
Restricted Access                    51   13.456464
Station Phone                        12    3.166227
EV Level2 EVSE Num                    9    2.374670
EV Network Web                        9    2.374670
Access Days Time                      5    1.319261
Date Last Confirmed                   1    0.263852
Groups With Access Code (French)      0    0.000000
Access Code                           0    0.000000
Country     

Eliminando 3 columnas por nombre que no es indispensable para los datos

In [14]:
df.columns

Index(['Fuel Type Code', 'Station Name', 'Street Address',
       'Intersection Directions', 'City', 'State', 'ZIP', 'Station Phone',
       'Status Code', 'Groups With Access Code', 'Access Days Time',
       'Cards Accepted', 'EV Level2 EVSE Num', 'EV DC Fast Count',
       'EV Network', 'EV Network Web', 'Geocode Status', 'Latitude',
       'Longitude', 'Date Last Confirmed', 'ID', 'Updated At',
       'Owner Type Code', 'Federal Agency ID', 'Federal Agency Name',
       'Open Date', 'EV Connector Types', 'Country',
       'Groups With Access Code (French)', 'Access Code', 'Access Detail Code',
       'Federal Agency Code', 'Facility Type', 'EV Pricing',
       'Restricted Access'],
      dtype='object')

In [15]:
df = df.drop(['Federal Agency Code', 'Federal Agency ID', 'Federal Agency Name', 'Date Last Confirmed'], axis=1)

In [16]:
df

Fuel Type Code                               Station Name  \
7777            ELEC                Port Authority Bus Terminal   
9645            ELEC                      United Parcel Service   
14384           ELEC                                   Spectrum   
14463           ELEC                 Central Parking - Tower 31   
14669           ELEC                               Tribeca Park   
...              ...                                        ...   
69770           ELEC                             DOCK STATION 9   
69771           ELEC                            DOCK STATION 10   
69823           ELEC                      Lincoln Square Garage   
69940           ELEC                              320 Mason Ave   
70215           ELEC  Delancey & Essex Municipal Parking Garage   

          Street Address                            Intersection Directions  \
7777         625 8th Ave                                                NaN   
9645       643 W 43rd St                                                NaN   
14384     401 W 219th St                                                NaN   
14463    9 W 31st Street                                                NaN   
14669    400 Chambers St  Residents & Visitors Only; See valet \nattenda...   
...                  ...                                                ...   
69770      1540 Broadway                                                NaN   
69771      1540 Broadway                                                NaN   
69823  178 W 69th Street  Entrance on 179 W 68th St or 184 W 69th \nSt b...   
69940      320 Mason Ave                                                NaN   
70215        107 Essx St                                                NaN   

           City State    ZIP Station Phone Status Code  \
7777   New York    NY  10018  866-816-7584           E   
9645   New York    NY  10036           NaN           E   
14384  New York    NY  10034  866-816-7584           E   
14463  New York    NY  10001  888-998-2546           E   
14669  New York    NY  10282  888-998-2546           E   
...         ...   ...    ...           ...         ...   
69770  New York    NY  10155  888-758-4389           E   
69771  New York    NY  10001  888-758-4389           E   
69823  New York    NY  10023  888-998-2546           E   
69940  New York    NY  10305  888-356-8911           E   
70215  New York    NY  10002  866-816-7584           E   

                 Groups With Access Code  ... Owner Type Code   Open Date  \
7777                              Public  ...               J  2018-04-01   
9645                             Private  ...               P  2016-03-01   
14384                             Public  ...               P  2016-06-27   
14463                             Public  ...             NaN  2014-08-20   
14669                            Private  ...             NaN  2014-11-12   
...                                  ...  ...             ...         ...   
69770                             Public  ...             NaN  2022-07-12   
69771                             Public  ...             NaN  2022-07-12   
69823                             Public  ...             NaN  2022-07-12   
69940                             Public  ...             NaN  2022-07-15   
70215  Public - Credit card at all times  ...               P  2022-07-16   

       EV Connector Types  Country        Groups With Access Code (French)  \
7777                J1772       US                                  Public   
9645                J1772       US                                   Privé   
14384               J1772       US                                  Public   
14463               J1772       US                                  Public   
14669               J1772       US                                   Privé   
...                   ...      ...                                     ...   
69770               J1772       US                                  Public   
69771 

In [17]:
can_nulos()

                                  Nulos  Porcentaje
EV DC Fast Count                    370   97.625330
Access Detail Code                  364   96.042216
Cards Accepted                      347   91.556728
Intersection Directions             347   91.556728
Facility Type                       119   31.398417
Owner Type Code                     100   26.385224
EV Pricing                           88   23.218997
Restricted Access                    51   13.456464
Station Phone                        12    3.166227
EV Level2 EVSE Num                    9    2.374670
EV Network Web                        9    2.374670
Access Days Time                      5    1.319261
Groups With Access Code               0    0.000000
Street Address                        0    0.000000
City                                  0    0.000000
State                                 0    0.000000
Access Code                           0    0.000000
Groups With Access Code (French)      0    0.000000
Country     

In [18]:
df.columns

Index(['Fuel Type Code', 'Station Name', 'Street Address',
       'Intersection Directions', 'City', 'State', 'ZIP', 'Station Phone',
       'Status Code', 'Groups With Access Code', 'Access Days Time',
       'Cards Accepted', 'EV Level2 EVSE Num', 'EV DC Fast Count',
       'EV Network', 'EV Network Web', 'Geocode Status', 'Latitude',
       'Longitude', 'ID', 'Updated At', 'Owner Type Code', 'Open Date',
       'EV Connector Types', 'Country', 'Groups With Access Code (French)',
       'Access Code', 'Access Detail Code', 'Facility Type', 'EV Pricing',
       'Restricted Access'],
      dtype='object')

In [19]:
df = df.drop(['Access Detail Code', 'Cards Accepted', 'Intersection Directions', 'Owner Type Code', 'Groups With Access Code (French)', 'Access Days Time', 'ZIP', 'Station Phone'], axis=1)

In [20]:
df

Fuel Type Code                               Station Name  \
7777            ELEC                Port Authority Bus Terminal   
9645            ELEC                      United Parcel Service   
14384           ELEC                                   Spectrum   
14463           ELEC                 Central Parking - Tower 31   
14669           ELEC                               Tribeca Park   
...              ...                                        ...   
69770           ELEC                             DOCK STATION 9   
69771           ELEC                            DOCK STATION 10   
69823           ELEC                      Lincoln Square Garage   
69940           ELEC                              320 Mason Ave   
70215           ELEC  Delancey & Essex Municipal Parking Garage   

          Street Address      City State Status Code  \
7777         625 8th Ave  New York    NY           E   
9645       643 W 43rd St  New York    NY           E   
14384     401 W 219th St  New York    NY           E   
14463    9 W 31st Street  New York    NY           E   
14669    400 Chambers St  New York    NY           E   
...                  ...       ...   ...         ...   
69770      1540 Broadway  New York    NY           E   
69771      1540 Broadway  New York    NY           E   
69823  178 W 69th Street  New York    NY           E   
69940      320 Mason Ave  New York    NY           E   
70215        107 Essx St  New York    NY           E   

                 Groups With Access Code  EV Level2 EVSE Num  \
7777                              Public                 2.0   
9645                             Private                 2.0   
14384                             Public                 8.0   
14463                             Public                 1.0   
14669                            Private                 2.0   
...                                  ...                 ...   
69770                             Public                 1.0   
69771                             Public                 1.0   
69823                             Public                 1.0   
69940                             Public                 4.0   
70215  Public - Credit card at all times                 NaN   

       EV DC Fast Count           EV Network  ...  Longitude      ID  \
7777                NaN           EV Connect  ... -73.990558   65287   
9645                NaN        Non-Networked  ... -74.000055   74648   
14384               NaN           EV Connect  ... -73.911616   94854   
14463               NaN        Blink Network  ... -73.986641   95069   
14669               NaN        Blink Network  ... -74.015049   96646   
...                 ...                  ...  ...        ...     ...   
69770               NaN  ChargePoint Network  ... -73.968188  223322   
69771               NaN  ChargePoint Network  ... -73.984391  223323   
69823               NaN        Blink Network  ... -73.982820  223392   
69940               NaN                  FLO  ... -74.088044  223526   
70215               4.0           EV Connect  ... -73.988197  224172   

                    Updated At   Open Date  EV Connector Types Country  \
7777   2021-03-11 23:22:17 UTC  2018-04-01               J1772      US   
9645   2022-06-14 16:24:51 UTC  2016-03-01               J1772      US   
14384  2022-04-28 20:01:31 UTC  2016-06-27               J1772      US   
14463  2022-07-24 22:55:19 UTC  2014-08-20               J1772      US   
14669  2022-07-24 22:55:19 UTC  2014-11-12               J1772      US   
...                        ...         ...                 ...     ...   
69770  2022-07-25 01:27:05 UTC  2022-07-12               J1772      US   
69771  2022-07-25 01:27:05 UTC  2022-07-12               J1772      US   
69823  2022-07-24 22:55:19 UTC  2022-07-12               J1772      US   
69940  2022-07-25 01:48:14 UTC  2022-07-15               J1772      US   
70215  2022-07-21 22:42:07 UTC  2022-07-16  CHADEMO J1772COMBO      US   

     

In [21]:
can_nulos()

                         Nulos  Porcentaje
EV DC Fast Count           370   97.625330
Facility Type              119   31.398417
EV Pricing                  88   23.218997
Restricted Access           51   13.456464
EV Level2 EVSE Num           9    2.374670
EV Network Web               9    2.374670
Longitude                    0    0.000000
Access Code                  0    0.000000
Country                      0    0.000000
EV Connector Types           0    0.000000
Open Date                    0    0.000000
Updated At                   0    0.000000
ID                           0    0.000000
Fuel Type Code               0    0.000000
Latitude                     0    0.000000
Station Name                 0    0.000000
EV Network                   0    0.000000
Groups With Access Code      0    0.000000
Status Code                  0    0.000000
State                        0    0.000000
City                         0    0.000000
Street Address               0    0.000000
Geocode Sta

In [22]:
df['EV DC Fast Count'].unique()

array([nan,  2.,  4.,  1.])

In [23]:
df['EV DC Fast Count'] = df['EV DC Fast Count'].fillna(0)

In [24]:
df

Fuel Type Code                               Station Name  \
7777            ELEC                Port Authority Bus Terminal   
9645            ELEC                      United Parcel Service   
14384           ELEC                                   Spectrum   
14463           ELEC                 Central Parking - Tower 31   
14669           ELEC                               Tribeca Park   
...              ...                                        ...   
69770           ELEC                             DOCK STATION 9   
69771           ELEC                            DOCK STATION 10   
69823           ELEC                      Lincoln Square Garage   
69940           ELEC                              320 Mason Ave   
70215           ELEC  Delancey & Essex Municipal Parking Garage   

          Street Address      City State Status Code  \
7777         625 8th Ave  New York    NY           E   
9645       643 W 43rd St  New York    NY           E   
14384     401 W 219th St  New York    NY           E   
14463    9 W 31st Street  New York    NY           E   
14669    400 Chambers St  New York    NY           E   
...                  ...       ...   ...         ...   
69770      1540 Broadway  New York    NY           E   
69771      1540 Broadway  New York    NY           E   
69823  178 W 69th Street  New York    NY           E   
69940      320 Mason Ave  New York    NY           E   
70215        107 Essx St  New York    NY           E   

                 Groups With Access Code  EV Level2 EVSE Num  \
7777                              Public                 2.0   
9645                             Private                 2.0   
14384                             Public                 8.0   
14463                             Public                 1.0   
14669                            Private                 2.0   
...                                  ...                 ...   
69770                             Public                 1.0   
69771                             Public                 1.0   
69823                             Public                 1.0   
69940                             Public                 4.0   
70215  Public - Credit card at all times                 NaN   

       EV DC Fast Count           EV Network  ...  Longitude      ID  \
7777                0.0           EV Connect  ... -73.990558   65287   
9645                0.0        Non-Networked  ... -74.000055   74648   
14384               0.0           EV Connect  ... -73.911616   94854   
14463               0.0        Blink Network  ... -73.986641   95069   
14669               0.0        Blink Network  ... -74.015049   96646   
...                 ...                  ...  ...        ...     ...   
69770               0.0  ChargePoint Network  ... -73.968188  223322   
69771               0.0  ChargePoint Network  ... -73.984391  223323   
69823               0.0        Blink Network  ... -73.982820  223392   
69940               0.0                  FLO  ... -74.088044  223526   
70215               4.0           EV Connect  ... -73.988197  224172   

                    Updated At   Open Date  EV Connector Types Country  \
7777   2021-03-11 23:22:17 UTC  2018-04-01               J1772      US   
9645   2022-06-14 16:24:51 UTC  2016-03-01               J1772      US   
14384  2022-04-28 20:01:31 UTC  2016-06-27               J1772      US   
14463  2022-07-24 22:55:19 UTC  2014-08-20               J1772      US   
14669  2022-07-24 22:55:19 UTC  2014-11-12               J1772      US   
...                        ...         ...                 ...     ...   
69770  2022-07-25 01:27:05 UTC  2022-07-12               J1772      US   
69771  2022-07-25 01:27:05 UTC  2022-07-12               J1772      US   
69823  2022-07-24 22:55:19 UTC  2022-07-12               J1772      US   
69940  2022-07-25 01:48:14 UTC  2022-07-15               J1772      US   
70215  2022-07-21 22:42:07 UTC  2022-07-16  CHADEMO J1772COMBO      US   

     

Remplazar los valores si contiene la palabra public

In [25]:
can_nulos()

                         Nulos  Porcentaje
Facility Type              119   31.398417
EV Pricing                  88   23.218997
Restricted Access           51   13.456464
EV Level2 EVSE Num           9    2.374670
EV Network Web               9    2.374670
Longitude                    0    0.000000
Access Code                  0    0.000000
Country                      0    0.000000
EV Connector Types           0    0.000000
Open Date                    0    0.000000
Updated At                   0    0.000000
ID                           0    0.000000
Fuel Type Code               0    0.000000
Latitude                     0    0.000000
Station Name                 0    0.000000
EV Network                   0    0.000000
EV DC Fast Count             0    0.000000
Groups With Access Code      0    0.000000
Status Code                  0    0.000000
State                        0    0.000000
City                         0    0.000000
Street Address               0    0.000000
Geocode Sta

In [26]:
df['Facility Type'].unique()

array(['PARKING_GARAGE', 'FLEET_GARAGE', 'OFFICE_BLDG', nan, 'FED_GOV',
       'PAY_GARAGE', 'HOTEL', 'PARKING_LOT', 'COLLEGE_CAMPUS', 'HOSPITAL'],
      dtype=object)

Cambiando nan por 'Sin Dato'

In [27]:
df['Facility Type'] = df['Facility Type'].fillna('Sin Dato')

In [28]:
can_nulos()

                         Nulos  Porcentaje
EV Pricing                  88   23.218997
Restricted Access           51   13.456464
EV Level2 EVSE Num           9    2.374670
EV Network Web               9    2.374670
Longitude                    0    0.000000
Facility Type                0    0.000000
Access Code                  0    0.000000
Country                      0    0.000000
EV Connector Types           0    0.000000
Open Date                    0    0.000000
Updated At                   0    0.000000
ID                           0    0.000000
Fuel Type Code               0    0.000000
Latitude                     0    0.000000
Station Name                 0    0.000000
EV Network                   0    0.000000
EV DC Fast Count             0    0.000000
Groups With Access Code      0    0.000000
Status Code                  0    0.000000
State                        0    0.000000
City                         0    0.000000
Street Address               0    0.000000
Geocode Sta

In [29]:
df['EV Pricing'] = df['EV Pricing'].fillna('Sin Dato')

In [30]:
can_nulos()

                         Nulos  Porcentaje
Restricted Access           51   13.456464
EV Level2 EVSE Num           9    2.374670
EV Network Web               9    2.374670
Longitude                    0    0.000000
EV Pricing                   0    0.000000
Facility Type                0    0.000000
Access Code                  0    0.000000
Country                      0    0.000000
EV Connector Types           0    0.000000
Open Date                    0    0.000000
Updated At                   0    0.000000
ID                           0    0.000000
Fuel Type Code               0    0.000000
Latitude                     0    0.000000
Station Name                 0    0.000000
EV Network                   0    0.000000
EV DC Fast Count             0    0.000000
Groups With Access Code      0    0.000000
Status Code                  0    0.000000
State                        0    0.000000
City                         0    0.000000
Street Address               0    0.000000
Geocode Sta

In [31]:
df['Restricted Access'] = df['Restricted Access'].fillna(True)

C:\Users\Usuario\AppData\Local\Temp\ipykernel_17636\566196007.py:1: FutureWarning: Downcasting object dtype arrays on .fillna, .ffill, .bfill is deprecated and will change in a future version. Call result.infer_objects(copy=False) instead. To opt-in to the future behavior, set `pd.set_option('future.no_silent_downcasting', True)`
  df['Restricted Access'] = df['Restricted Access'].fillna(True)


In [32]:
can_nulos()

                         Nulos  Porcentaje
EV Level2 EVSE Num           9     2.37467
EV Network Web               9     2.37467
Fuel Type Code               0     0.00000
Longitude                    0     0.00000
EV Pricing                   0     0.00000
Facility Type                0     0.00000
Access Code                  0     0.00000
Country                      0     0.00000
EV Connector Types           0     0.00000
Open Date                    0     0.00000
Updated At                   0     0.00000
ID                           0     0.00000
Geocode Status               0     0.00000
Latitude                     0     0.00000
Station Name                 0     0.00000
EV Network                   0     0.00000
EV DC Fast Count             0     0.00000
Groups With Access Code      0     0.00000
Status Code                  0     0.00000
State                        0     0.00000
City                         0     0.00000
Street Address               0     0.00000
Restricted 

In [33]:
df['EV Level2 EVSE Num'] = df['EV Level2 EVSE Num'].fillna('Sin Dato')

In [34]:
df['EV Network Web'] = df['EV Network Web'].fillna('Sin Dato')

In [35]:
df

Fuel Type Code                               Station Name  \
7777            ELEC                Port Authority Bus Terminal   
9645            ELEC                      United Parcel Service   
14384           ELEC                                   Spectrum   
14463           ELEC                 Central Parking - Tower 31   
14669           ELEC                               Tribeca Park   
...              ...                                        ...   
69770           ELEC                             DOCK STATION 9   
69771           ELEC                            DOCK STATION 10   
69823           ELEC                      Lincoln Square Garage   
69940           ELEC                              320 Mason Ave   
70215           ELEC  Delancey & Essex Municipal Parking Garage   

          Street Address      City State Status Code  \
7777         625 8th Ave  New York    NY           E   
9645       643 W 43rd St  New York    NY           E   
14384     401 W 219th St  New York    NY           E   
14463    9 W 31st Street  New York    NY           E   
14669    400 Chambers St  New York    NY           E   
...                  ...       ...   ...         ...   
69770      1540 Broadway  New York    NY           E   
69771      1540 Broadway  New York    NY           E   
69823  178 W 69th Street  New York    NY           E   
69940      320 Mason Ave  New York    NY           E   
70215        107 Essx St  New York    NY           E   

                 Groups With Access Code EV Level2 EVSE Num  EV DC Fast Count  \
7777                              Public                2.0               0.0   
9645                             Private                2.0               0.0   
14384                             Public                8.0               0.0   
14463                             Public                1.0               0.0   
14669                            Private                2.0               0.0   
...                                  ...                ...               ...   
69770                             Public                1.0               0.0   
69771                             Public                1.0               0.0   
69823                             Public                1.0               0.0   
69940                             Public                4.0               0.0   
70215  Public - Credit card at all times           Sin Dato               4.0   

                EV Network  ...  Longitude      ID               Updated At  \
7777            EV Connect  ... -73.990558   65287  2021-03-11 23:22:17 UTC   
9645         Non-Networked  ... -74.000055   74648  2022-06-14 16:24:51 UTC   
14384           EV Connect  ... -73.911616   94854  2022-04-28 20:01:31 UTC   
14463        Blink Network  ... -73.986641   95069  2022-07-24 22:55:19 UTC   
14669        Blink Network  ... -74.015049   96646  2022-07-24 22:55:19 UTC   
...                    ...  ...        ...     ...                      ...   
69770  ChargePoint Network  ... -73.968188  223322  2022-07-25 01:27:05 UTC   
69771  ChargePoint Network  ... -73.984391  223323  2022-07-25 01:27:05 UTC   
69823        Blink Network  ... -73.982820  223392  2022-07-24 22:55:19 UTC   
69940                  FLO  ... -74.088044  223526  2022-07-25 01:48:14 UTC   
70215           EV Connect  ... -73.988197  224172  2022-07-21 22:42:07 UTC   

        Open Date  EV Connector Types Country Access Code   Facility Type  \
7777   2018-04-01               J1772      US      public  PARKING_GARAGE   
9645   2016-03-01               J1772      US     private    FLEET_GARAGE   
14384  2016-06-27               J1772      US      public     OFFICE_BLDG   
14463  2014-08-20               J1772      US      public        Sin Dato   
14669  2014-11-12               J1772      US     private        Sin Dato   
...           ...                 ...     ...         ...             ...   
69770  2022-07-12               J1772      US      public        Si

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 379 entries, 7777 to 70215
Data columns (total 23 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Fuel Type Code           379 non-null    object 
 1   Station Name             379 non-null    object 
 2   Street Address           379 non-null    object 
 3   City                     379 non-null    object 
 4   State                    379 non-null    object 
 5   Status Code              379 non-null    object 
 6   Groups With Access Code  379 non-null    object 
 7   EV Level2 EVSE Num       379 non-null    object 
 8   EV DC Fast Count         379 non-null    float64
 9   EV Network               379 non-null    object 
 10  EV Network Web           379 non-null    object 
 11  Geocode Status           379 non-null    object 
 12  Latitude                 379 non-null    float64
 13  Longitude                379 non-null    float64
 14  ID                       3

In [37]:
df_fuel_charging = df

Para este archivo se realiza la carga de una tabla de  379 filas y 23 columnas, si ejecutas el código ya se ecunentran los datos filtrados y limpios, ademas de losque sóloserán de utilidad para el proyecto.

La tabla se ecnuetra guardada en un DataFrame llamado df_df_fuel_charging

Se guardan y exportan los datos limpios en un archivo csv dentro de la carpeta Clean Data en la carpeta Datasets

In [38]:
df_fuel_charging.to_csv('../Datasets/Clean Data/fuel_charging.csv', index=False)